### PR TITLE
feat: earth day challenge

### PR DESCRIPTION
**Pull request by eddieferrer**
* basically the same as IWD challenge
* Added badge to lending stats (screenshot)
* Moved achievement key's into constants in util
* Includes thank you page (see screenshot)
* Includes redirect from checkout page
* Challenge dates prod (4/22-4/24)
* Challenge dates dev (4/3-4/24)

Screen Shot 2023-04-06 at 12 40 48 PM
Screen Shot 2023-04-04 at 11 29 54 AM

### CODE REVIEW

File: src/pages/Thanks/ThanksPageEarthDayChallenge.vue [hunk 1]
```diff
@@ -189,34 +187,39 @@ export default {
 					visitorId: cookieStore.get('uiv') || null,
 				}
 			}).then(({ data }) => {
+				console.log(data);
+				console.log('userId', data?.my?.userAccount?.id);
```
Comment (dyersituations) [inline src/pages/Thanks/ThanksPageEarthDayChallenge.vue:191]: Remove 2 logs

File: src/pages/Thanks/ThanksPageEarthDayChallenge.vue [hunk 2]
```diff
@@ -189,34 +187,39 @@ export default {
...
 				return client.query({
 					query: userAchievementsProgress,
 					variables: {
-						userId: data?.my?.userAccount?.id,
+						userId: data?.my?.userAccount?.id.toString(),
```
Comment (dyersituations) [inline src/pages/Thanks/ThanksPageEarthDayChallenge.vue:195]: data?.my?.userAccount?.id?.toString()

File: src/pages/Thanks/ThanksPageEarthDayChallenge.vue [hunk 3]
```diff
@@ -277,6 +267,21 @@ export default {
...
 			);
 		}
+
+		// Get achievement completion status
+		let achievementData = {};
+		try {
+			achievementData = this.apollo.readQuery({
+				query: userAchievementsProgress,
+				variables: {
+					userId: data?.my?.userAccount?.id.toString(),
```
Comment (dyersituations) [inline src/pages/Thanks/ThanksPageEarthDayChallenge.vue:277]: data?.my?.userAccount?.id?.toString()

**Review by dyersituations — APPROVED** [review]
I'm guessing long-term we'll want to abstract out the thanks page for any campaign/event w/ badge.

File: src/pages/Thanks/ThanksPageEarthDayChallenge.vue (hunk 1 above)
Comment (eddieferrer) [reply src/pages/Thanks/ThanksPageEarthDayChallenge.vue:191]: > Remove 2 logs

Whoops!

**Comment by eddieferrer** [comment]
> I'm guessing long-term we'll want to abstract out the thanks page for any campaign/event w/ badge.

Yeah probably. It would be nice if marketing could just set up these challenges with minimal engineering help, but it has been unclear how much we should invest in the challenges

**Review by mcstover — APPROVED** [review]
I'm also very interested to see this rolled into the standard thanks page, maybe with contentful control for the acheivement content...

**Pull request merged**